src/modules/universal-discovery/components/create/create.choose.content.type.component.js

```diff
@@ -7,19 +7,50 @@ export default class ChooseContentTypeComponent extends Component {
     constructor(props) {
         super(props);
 
+        let selected = {};
+        const isForcedContentType = props.allowedContentTypes.length === 1;
+        const allowedIncludesPreselected = props.allowedContentTypes.includes(props.preselectedContentType);
+        const isPreselectedContentType = props.preselectedContentType && (!props.allowedContentTypes.length || allowedIncludesPreselected);
```

Comment:
I would move this check: `!props.allowedContentTypes.length` to the `hasPreselectedAllowedContentType` variable.